tests/ui/associated-inherent-types/former-subst-ice.rs

```diff
@@ -0,0 +1,16 @@
+// check-pass
```

Comment:
Not sure if this test is still useful since it's kind of superseded by `inference.rs`. I've kept it solely to explicitly demonstrate that `bugs/ice-substitution.rs` is fixed in this PR.